### DISCUSSION TITLE
Slow DELETE statement (no where clause) - Is it expected?

### DISCUSSION THREAD

**Discussion by fedehann** [duckdb/duckdb#5894] [General]
Hi!
Probably a dumb question.
I was just running some experiments with DuckDb and I noticed that DELETING rows from huge tables (with no filters) takes forever. Is this expected? Should I recreate every table from scratch instead of truncating it and re-populating the latter?

Thanks in advance for any input!

**Comment by Mytherin** [comment]
We currently don't have any specific optimizations for deleting everything in a table - dropping and recreating the table will be more efficient.

**Reply by fedehann** [reply]
Ok, thanks! Is it normal, being there no optimization at all for truncating-like DELETE, for the operation to take many times more than the INSERT operation (from a CSV). (At least 10 times more, than I got tired and stop that)

**Reply by Mytherin** [reply]
We have done a lot more optimizations around insertions than around deletions. There is likely room for improvement there. I would say for now it is expected that deletions are slower.
